src/libraries/System.IO.Compression/tests/CompressionStreamUnitTests.Gzip.cs

```diff
@@ -153,6 +155,68 @@ public async Task ManyManyConcatenatedGzipStreams(int streamCount, TestScenario
             await TestConcatenatedGzipStreams(streamCount, scenario, bufferSize, bytesPerStream);
         }
 
+        [Theory]
+        [InlineData(TestScenario.ReadAsync)]
+        [InlineData(TestScenario.Read)]
+        [InlineData(TestScenario.Copy)]
+        [InlineData(TestScenario.CopyAsync)]
+        public async Task StreamTruncation_IsDetected(TestScenario scenario)
+        {
+            var source = Enumerable.Range(0, 32).Select(i => (byte)i).ToArray();
+            byte[] data;
+            using (var compressed = new MemoryStream())
+            using (var gzip = new GZipStream(compressed, CompressionLevel.Fastest))
+            {
+                foreach (var b in source)
+                {
+                    gzip.WriteByte(b);
+                }
+
+                gzip.Dispose();
+                data = compressed.ToArray();
+            }
+
+            for (var i = 1; i < data.Length; i++)
```

Comment:
does this manage to hit all the 4 new throw statements?

Reply:
Yes it does.